plenum/server/request_managers/write_request_manager.py

```diff
@@ -314,19 +315,25 @@ def do_taa_validation(self, request: Request, req_pp_time: int, config):
             )
 
         r_taa_a_ts = request.taaAcceptance[f.TAA_ACCEPTANCE_TIME.nm]
-        ts_lowest = (
+        datetime_r_taa = datetime.utcfromtimestamp(r_taa_a_ts)
+        if datetime_r_taa.time() != time(0):
+            raise InvalidClientTaaAcceptanceError(
+                request.identifier, request.reqId,
+                "TAA timestamp {} is too precise and is a privacy risk."
```

Comment:
I think it would be more clear to have the following error message (to be more consistent with other error messages):
`Txn Author Agreement acceptance time {} is too precise and is a privacy risk.`

Reply:
Will be changed in https://github.com/hyperledger/indy-plenum/pull/1263